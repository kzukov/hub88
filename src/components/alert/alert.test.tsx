import Alert from "./alert";
import { render, screen } from "@testing-library/react";

describe("Alert", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<Alert type="primary">children</Alert>);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Structure", () => {
    const testChildren = "Test children";
    const testType = "primary";

    const { container } = render(<Alert type={testType}>{testChildren}</Alert>);

    expect(screen.getByText(testChildren)).toBeInTheDocument();
    expect(container.firstChild).toHaveClass(`alert alert-${testType}`);
  });
});
