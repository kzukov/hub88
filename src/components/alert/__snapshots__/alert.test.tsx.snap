// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshot 1`] = `
<DocumentFragment>
  <div
    class="alert alert-primary"
    role="alert"
  >
    children
  </div>
</DocumentFragment>
`;
